internal/helm/envresolver.go

```diff
@@ -60,38 +66,71 @@ type EnvResolver interface {
 	ResolvePlatformEnv(ctx context.Context) (HelmEnv, error)
 }
 
-type helmEnvResolver struct {
+type helm2EnvResolver struct {
 	// helmHomes the configurable directory location where helm homes are to be set up
 	helmHomes  string
 	orgService OrgService
 	logger     Logger
 }
 
-func NewHelmEnvResolver(helmHome string, orgService OrgService, logger Logger) EnvResolver {
-	return helmEnvResolver{
+func NewHelm2EnvResolver(helmHome string, orgService OrgService, logger Logger) EnvResolver {
+	return helm2EnvResolver{
 		helmHomes:  helmHome,
 		orgService: orgService,
 		logger:     logger,
 	}
 }
 
-func (h helmEnvResolver) ResolveHelmEnv(ctx context.Context, organizationID uint) (HelmEnv, error) {
-	h.logger.Debug("resolving organization helm env home")
-	orgName, err := h.orgService.GetOrgNameByOrgID(ctx, organizationID)
+func (h2r helm2EnvResolver) ResolveHelmEnv(ctx context.Context, organizationID uint) (HelmEnv, error) {
+	h2r.logger.Debug("resolving organization helm env home")
+	orgName, err := h2r.orgService.GetOrgNameByOrgID(ctx, organizationID)
 	if err != nil {
 		return HelmEnv{}, errors.WrapIfWithDetails(err, "failed to get organization name for ID",
 			"organizationID", organizationID)
 	}
 
 	return HelmEnv{
-		home:     path.Join(h.helmHomes, orgName, helmPostFix),
+		home:     path.Join(h2r.helmHomes, orgName, helmPostFix),
 		platform: false,
 	}, nil
 }
 
-func (h helmEnvResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
+func (h2r helm2EnvResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
 	return HelmEnv{
-		home:     path.Join(h.helmHomes, PlatformHelmHome, helmPostFix),
+		home:     path.Join(h2r.helmHomes, PlatformHelmHome, helmPostFix),
 		platform: true,
 	}, nil
 }
+
+// helm3EnvResolver helm env resolver to be used for resolving helm 3 environments
+type helm3EnvResolver struct {
+	delegate EnvResolver
+}
+
+func NewHelm3EnvResolver(delegate EnvResolver) EnvResolver {
```

Comment:
Reusing helm 2 code doesn't make much sense here as we want to delete it at some point.

Reply:
sure, this seems overcomplicated (i didnt want to duplicat logic, will fix it)